crypto/ec/ec_lib.c

```diff
@@ -811,16 +812,33 @@ int EC_POINT_set_Jprojective_coordinates_GFp(const EC_GROUP *group,
               EC_R_INCOMPATIBLE_OBJECTS);
         return 0;
     }
-    return group->meth->point_set_Jprojective_coordinates_GFp(group, point, x,
-                                                              y, z, ctx);
+    if (x == NULL || y == NULL || z == NULL) {
+        ECerr(EC_F_EC_POINT_SET_JPROJECTIVE_COORDINATES_GFP,
+              ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+
+    if (!BN_nnmod(point->X, x, group->field, ctx)
+        || !BN_nnmod(point->Y, y, group->field, ctx)
+        || !BN_nnmod(point->Z, z, group->field, ctx)
+        || (group->meth->field_encode != NULL
+            && (!group->meth->field_encode(group, point->X, point->X, ctx)
+                || !group->meth->field_encode(group, point->Y, point->Y, ctx)
+                || !group->meth->field_encode(group, point->Z, point->Z, ctx)))) {
+        ECerr(EC_F_EC_POINT_SET_JPROJECTIVE_COORDINATES_GFP, ERR_R_INTERNAL_ERROR);
+        return 0;
+    }
+
+    point->Z_is_one = 0;
+    return 1;
```

Comment:
This was early on in the change, and now seems a bit heavy handed. Should I just call `ec_GFp_simple_set_Jprojective_coordinates_GFp` directly here?